sd_webui_bayesian_merger/merger.py

```diff
@@ -85,30 +161,37 @@ def load_sd_model(self, model_path: PathT) -> SDModel:
     def merge_key(
         self,
         key: str,
-        weights_alpha: List[float],
-        weights_beta: List[float],
-        base_alpha: float,
-        base_beta: float,
+        thetas: Dict,
+        weights: Dict,
+        bases: Dict,
         best: bool,
-        theta_0: Dict,
-        theta_1: Dict,
-        theta_2: Dict,
     ) -> Tuple[str, Dict]:
-        if "model" not in key or key not in theta_1:
-            return
-        if theta_2 and key not in theta_2:
-            return
         if KEY_POSITION_IDS in key:
-            return 
-
-        re_inp = re.compile(r"\.input_blocks\.(\d+)\.")  # 12
-        re_mid = re.compile(r"\.middle_block\.(\d+)\.")  # 1
-        re_out = re.compile(r"\.output_blocks\.(\d+)\.")  # 12
-        c_alpha = base_alpha
-        c_beta = base_beta
+            if self.cfg.skip_position_ids == 1:
+                if not best or self.cfg.best_precision == "16":
+                    return (key, thetas["model_a"][key].half())
+                return (
+                    key,
+                    thetas["model_a"][key],
+                )  # Skip position_ids key to eject effect. Value of Model A used.
+            elif self.cfg.skip_position_ids == 2:
+                thetas["model_a"][key] = torch.tensor(
+                    [list(range(MAX_TOKENS))], dtype=torch.int64
+                )
+                if not best or self.cfg.best_precision == "16":
+                    return (key, thetas["model_a"][key].half())
+                return (key, thetas["model_a"][key])
```

Comment:
I refactored it according to MBW, which the one I was basing off of id 0 just didn't change it and left the fate of the key to the merger, so that's why there isn't a return here or a continue in the other one.